src/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -313,7 +298,7 @@ private void OnSerializing(StreamingContext ctx)
             // OnSerializing is called before serialization of an object
             try
             {
-                SetBits(GetBits(this));
+                new decimal(lo, mid, hi, flags);
```

Comment:
I think I am missing something obvious here. How does just creating a new decimal instance address anything? The previous code seems to have done a get/set of values on `this` instance where as now we create a new instance that is just collected - isn't this just a no-op?

Reply:
Oh... i see now. Wow this is rather convoluted. Would it be possible to just have a simple `ValidateParts()` static function that either returned `true`/`false` which then we could throw a specific exception? The private constructor just for validation seems weird.

Reply:
Sad 😞 Okay, well thanks for considering. Can you add an explicit comment in both locations indicating the reason they type is being created is to validate the arguments?